platform/iphone/in_app_store.mm

```diff
@@ -116,167 +92,150 @@ - (void)productsRequest:(SKProductsRequest *)request didReceiveResponse:(SKProdu
-
-	ProductsDelegate *delegate = [[ProductsDelegate alloc] init];
-
-	request.delegate = delegate;
-	[request start];
-
-	return OK;
-};
-
-Error InAppStore::restore_purchases() {
-
-	printf("restoring purchases!\n");
-	[[SKPaymentQueue defaultQueue] restoreCompletedTransactions];
-
-	return OK;
-};
-
-@interface TransObserver : NSObject <SKPaymentTransactionObserver> {
-};
-@end
-
-@implementation TransObserver
-
+#pragma mark - SKPaymentTransactionObserver
 - (void)paymentQueue:(SKPaymentQueue *)queue updatedTransactions:(NSArray *)transactions {
-
-	printf("transactions updated!\n");
 	for (SKPaymentTransaction *transaction in transactions) {
+		// Create a variant with this info
+		Dictionary ret = TransactionDictionary(transaction);
+		String pid = String::utf8([transaction.payment.productIdentifier UTF8String]);
 
 		switch (transaction.transactionState) {
 			case SKPaymentTransactionStatePurchased: {
 				printf("status purchased!\n");
-				String pid = String::utf8([transaction.payment.productIdentifier UTF8String]);
-				String transactionId = String::utf8([transaction.transactionIdentifier UTF8String]);
 				InAppStore::get_singleton()->_record_purchase(pid);
-				Dictionary ret;
-				ret["type"] = "purchase";
-				ret["result"] = "ok";
-				ret["product_id"] = pid;
-				ret["transaction_id"] = transactionId;
-
-				NSData *receipt = nil;
-				int sdk_version = 6;
-
-				if ([[[UIDevice currentDevice] systemVersion] floatValue] >= 7.0) {
-
-					NSURL *receiptFileURL = nil;
-					NSBundle *bundle = [NSBundle mainBundle];
-					if ([bundle respondsToSelector:@selector(appStoreReceiptURL)]) {
 
-						// Get the transaction receipt file path location in the app bundle.
-						receiptFileURL = [bundle appStoreReceiptURL];
+				// Get the transaction receipt file path location in the app
+				// bundle and read the contents
+				NSURL *receiptFileURL = [[NSBundle mainBundle] appStoreReceiptURL];
+				NSData *receipt = [NSData dataWithContentsOfURL:receiptFileURL];
 
-						// Read in the contents of the transaction file.
-						receipt = [NSData dataWithContentsOfURL:receiptFileURL];
-						sdk_version = 7;
-
-					} else {
-						// Fall back to deprecated transaction receipt,
-						// which is still available in iOS 7.
-
-						// Use SKPaymentTransaction's transactionReceipt.
-						receipt = transaction.transactionReceipt;
-					}
-
-				} else {
-					receipt = transaction.transactionReceipt;
-				}
-
-				NSString *receipt_to_send = nil;
-				if (receipt != nil) {
-					receipt_to_send = [receipt description];
-				}
+				// Send the receipt back to Godot
 				Dictionary receipt_ret;
-				receipt_ret["receipt"] = String::utf8(receipt_to_send != nil ? [receipt_to_send UTF8String] : "");
-				receipt_ret["sdk"] = sdk_version;
+				receipt_ret["receipt"] = String::utf8(receipt != nil ? [receipt.description UTF8String] : "");
+				receipt_ret["sdk"] = 7;
```

Comment:
For backwards compatibility I'm hard coding this to the most recent value that would have passed the version check.